src/chunk.c

```diff
@@ -1621,6 +1621,14 @@ chunk_tuple_found(TupleInfo *ti, void *arg)
 
 	chunk->relkind = get_rel_relkind(chunk->table_id);
 
+	if (!chunk->fd.osm_chunk)
+	{
+		chunk->amoid = ts_get_rel_am(chunk->table_id);
+		Assert(OidIsValid(chunk->amoid));
+	}
+	else
+		chunk->amoid = InvalidOid;
+
```

Comment:
Can you put this in the loop below that does `get_relkind`, and combine them into one `SearchSysCache` call? That would be just perfect, no new catalog lookups introduced in chunk expansion. Maybe it's convenient to repurpose `ts_get_rel_am` to get both and use it in the other place as well.

Reply:
I'll do that, but these are syscache lookups so the first call will at worst warm the cache for pg_class so that the next call will use the cached entry. It will at worst be a hash-table lookup. 

Building the chunk will only happen once and now we are already caching the chunk in the planner RelOptInfo so I struggle to understand how this can make much of a difference. We are building caches over caches.

Reply:
Yeah I understand, it still has a cost even on warm caches, and you start seeing it in profiles after 1k chunks. Not a very big one, but we already have the infrastructure in place to remove even this small cost with a small engineering effort, so I think it's good to do this.